sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingParserSuite.scala

```diff
@@ -1483,6 +1573,21 @@ class SqlScriptingParserSuite extends SparkFunSuite with SQLHelper {
     assert(whileStmt.label.contains("lbl"))
   }
 
+  test("loop with empty body") {
+    val sqlScriptText =
+      """BEGIN
+        | LOOP
+        | END LOOP;
+        |END
+      """.stripMargin
+    checkError(
+      exception = intercept[ParseException] {
+        parsePlan(sqlScriptText)
+      },
+      condition = "PARSE_SYNTAX_ERROR",
+      parameters = Map("error" -> "'LOOP'", "hint" -> ""))
+  }
+
   test("loop with if else block") {
```

Comment:
Does it also make sense to add test with for with empty body, in this PR or your other one, depending on which you merge first?

Reply:
Yes, I will add it to the one which gets merged last, or if they both get merged then I will create a separate PR.

Reply:
Added tests for FOR